src/components/ContentSidebar/DetailsSidebar.js

```diff
@@ -197,48 +215,69 @@ class DetailsSidebar extends React.PureComponent<Props, State> {
     }
 
     /**
-     * Sucess callback for classification change
+     * File classification fetch success callback.
      *
-     * @private
+     * @param {ClassificationInfo} classification - Info about the file's classification
      * @return {void}
      */
-    classifiationChangeSuccessCallback = (file: BoxItem) => {
+    fetchClassificationSuccessCallback = (classification: ClassificationInfo): void => {
         this.setState({
-            file,
-            isLoading: false,
+            classification,
+            classificationError: undefined,
+            isLoadingClassification: false,
         });
     };
 
     /**
-     * Error callback for classification change
+     * Handles a failed file classification fetch
      *
      * @private
+     * @param {ElementsXhrError} error - API error
      * @return {void}
      */
-    classifiationChangeErrorCallback = () => {
+    fetchClassificationErrorCallback = (error: ElementsXhrError): void => {
+        let classificationError;
+
+        if (getProp(error, 'status') === HTTP_STATUS_CODE_FORBIDDEN) {
+            classificationError = {
+                error: messages.fileClassificationPermissionsError,
+            };
+        } else {
+            classificationError = {
+                maskError: {
+                    errorHeader: messages.fileClassificationErrorHeaderMessage,
+                    errorSubHeader: messages.defaultErrorMaskSubHeaderMessage,
+                },
+            };
+        }
+
         this.setState({
-            isLoading: false,
+            classification: undefined,
+            classificationError,
+            isLoadingClassification: false,
+        });
+
+        this.props.onError(error, ERROR_CODE_FETCH_CLASSIFICATION, {
```

Comment:
In #661 I am changing each API to return the error code, not the component. Can you follow this pattern?

`(error: ElementsXhrError, code: string)`

Reply:
done!